### ISSUE TITLE
[BUG] QWEN 在 finetune 时没有将 label 错位

### ISSUE THREAD

**Issue by SefaZeng** [QwenLM/Qwen#834]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

在使用 Trainer 进行 finetune 的时候，Trainer 计算 loss 的逻辑是：
```
            if model_name in MODEL_FOR_CAUSAL_LM_MAPPING_NAMES.values():
                loss = self.label_smoother(outputs, labels, shift_labels=True)
            else:
                loss = self.label_smoother(outputs, labels)
```
因为 QWEN 的模型不在这个列表里面，所以 label 没有进行错位，导致每个 token 的 label 变成了自己。

### 期望行为 | Expected Behavior

label 正常错位，进行训练。

### 运行环境 | Environment

```Markdown
- OS: Linux
- Python: 3.8.11
- Transformers: 4.33.0
- PyTorch: 2.0.0
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`): 11.7
```

### 备注 | Anything else?

相关 issue: [QWEN-1.8B 使用非官方脚本 finetune 之后输出全是重复的 token](https://github.com/QwenLM/Qwen/issues/738)

**Comment by jklj077** [comment]
呃，这个比较tricky，你是一定用label smoothing吗？

我们的脚本里都是没有label smoothing的，所以Trainer代码不会走那个路径。

```python
def compute_loss(self, model, inputs, return_outputs=False):
    ...

    if self.label_smoother is not None and "labels" in inputs:
        labels = inputs.pop("labels")
    else:
        # 走的这里
        labels = None 
    outputs = model(**inputs)

    ...

    if labels is not None:
        ...
        if model_name in MODEL_FOR_CAUSAL_LM_MAPPING_NAMES.values():
            loss = self.label_smoother(outputs, labels, shift_labels=True)
        else:
            loss = self.label_smoother(outputs, labels)
    else:
        # 走的这里

        ...

        # modeling_qwen.py里算loss会对label移位
        loss = outputs["loss"] if isinstance(outputs, dict) else outputs[0]

    return (loss, outputs) if return_outputs else loss
```

**Comment by SefaZeng** [comment]
> 呃，这个比较tricky，你是一定用label smoothing吗？
> 
> 我们的脚本里都是没有label smoothing的，所以Trainer代码不会走那个路径。
> 
> ```python
> def compute_loss(self, model, inputs, return_outputs=False):
>     ...
> 
>     if self.label_smoother is not None and "labels" in inputs:
>         labels = inputs.pop("labels")
>     else:
>         # 走的这里
>         labels = None 
>     outputs = model(**inputs)
> 
>     ...
> 
>     if labels is not None:
>         ...
>         if model_name in MODEL_FOR_CAUSAL_LM_MAPPING_NAMES.values():
>             loss = self.label_smoother(outputs, labels, shift_labels=True)
>         else:
>             loss = self.label_smoother(outputs, labels)
>     else:
>         # 走的这里
> 
>         ...
> 
>         # modeling_qwen.py里算loss会对label移位
>         loss = outputs["loss"] if isinstance(outputs, dict) else outputs[0]
> 
>     return (loss, outputs) if return_outputs else loss
> ```

懂了，多谢。
